api/api.py

```diff
@@ -58,10 +59,41 @@ def get_filter_queries(earthquake_filters: dict[str]) -> list[str]:
         res.append(f"t.type_value = '{event}'")
     if min_magnitude is not None:
         res.append(f"e.magnitude >= '{min_magnitude}'")
-    if continent is not None:
-        # TODO: implement api to get continent filter working
-        res.append("NOT YET IMPLEMENTED")
-    res[0] = "WHERE "+res[0]
+    if res:
+        res[0] = "WHERE "+res[0]
+    return res
+
+
+def filter_by_continent(fetched_data, continent: str) -> list[dict]:
+    """filter through the data extracted so far, and only keep data
+    that matches the continent being filtered."""
+    continent_dict = {
+        "North America": "NA",
+        "South America": "SA",
+        "Asia": "AS",
+        "Africa": "AF",
+        "Oceania": "OC",
+        "Europe": "EU",
+        "Antarctica": "AQ"
+
+    }
+    res = []
+    if continent_dict.get(continent.title(), None) is None:
+        return fetched_data
+    continent_filter_code = continent_dict[continent.title()]
+    for row in fetched_data:
+        try:
+            location = rg.get((row["lat"], row["lon"]))
+            # extract country code
+            country_code = location["country_code"]
+            # get continent code from country code
+            continent_code = pc.country_alpha2_to_continent_code(country_code)
+            if continent_code == continent_filter_code:
+                res.append(row)
+                print("row added!")
+            print("run complete!")
```

Comment:
could use logging instead of prints statements maybe

Reply:
This was done for debugging, and should not be part of the final implementation. Thanks for pointing this out.